app/assets/javascripts/discourse/app/components/tag-groups-form.js

```diff
@@ -99,14 +99,18 @@ export default Component.extend(bufferedProperty("model"), {
     },
 
     setPermissionsGroups(groupIds) {
-      let permissions = {};
+      let updatedPermissions = Object.assign(
+        {},
+        this.buffered.get("permissions")
```

Comment:
It's worth noting that `Object.assign` doesn't do a deep copy. You'll only clone the top level keys this way. If the values are objects/arrays they will be references. If that's not what you want, you can use `cloneJSON` from `discourse-common/lib/object`